crates/hpos_connect_hc/src/app_connection.rs

```diff
@@ -34,36 +34,59 @@ pub struct AppConnection {
 }
 
 impl AppConnection {
+    // Connect to app interface for given installed app id if one already exists,
+    // otherwise attach a new app interface and then establish connection.
     pub async fn connect(
         admin_ws: &mut AdminWebsocket,
         keystore: MetaLairClient,
         app_id: String,
     ) -> Result<Self> {
-        let app_port = admin_ws
-            .attach_app_interface(None)
+        let attached_app_interfaces = admin_ws
+            .list_app_interfaces()
             .await
-            .context("failed to start app interface for core app")?;
+            .context("failed to start fetch app interfaces during app connection setup")?;
 
-        let token = admin_ws.issue_app_auth_token(app_id.clone()).await?;
+        let app_interface = attached_app_interfaces.into_iter().find(|a| {
+            a.installed_app_id.is_some() && a.installed_app_id.to_owned().unwrap() == app_id
+        });
+        println!(
+            "FOUND APP INTERFACE: >> ?? app_interface: {:?} ",
+            app_interface
+        );
+
+        let app_port = match app_interface {
+            Some(a) => a.port,
+            None => admin_ws
+                .attach_app_interface(None)
```

Comment:
If you are not assigning the port here to an `installed_app_id`  then the check above will never find a port with the `installed_app_id`

Reply:
Good call.  I didn't see that the installed_app_id assignment was optional.  I'll adjust that now.

Reply:
Done.